### PR TITLE
Allow underscores in DNS names from DHCP leases

### PR DESCRIPTION
**Pull request by bugfixin**
Currently DHCP leases with DNS names including underscores cause a warning "dhcpd leases: XXX_XXX not a valid hostname, ignoring" 

While these may not be true internet-valid hostnames according to IETF, many valid DNS names include underscores (e.g. _acme-challenge for ACME/letsencrypt validation, microsoft active directory names, SRV records, and manufacturer-default IP camera names, though I'm sure there are others). 

Blindly denying underscores for DNS records is likely to cause unforeseen issues, and I don't believe it's in our best interest to police names to that degree.

### CODE REVIEW

**Comment by AdSchellevis** [comment]
The current constraints might be a bit stricter then needed (introduced in https://github.com/opnsense/core/commit/02aae56f05f313e7a5bc59d1e2551864dbea9a12 due to https://github.com/opnsense/core/issues/6614), but someone does need to validate if allowing those will not lead to similar issues as those we started with. 

Strictly speaking rfc1123 does not allow underscores in hostnames if I'm not mistaken (https://datatracker.ietf.org/doc/html/rfc1123), so if the applications underneath (unbound/dnsmasq) are strict, they might lead to issues, in which case we should not try to feed those.

**Comment by bugfixin** [comment]
Per RFC 2181 section 11: https://www.rfc-editor.org/rfc/rfc2181#section-11

> Occasionally it is assumed that the Domain Name System serves only
   the purpose of mapping Internet host names to data, and mapping
   Internet addresses to host names.  This is not correct, the DNS is a
   general (if somewhat limited) hierarchical database, and can store
   almost any kind of data, for almost any purpose.

 >  The DNS itself places only one restriction on the particular labels
   that can be used to identify resource records.  That one restriction
   relates to the length of the label and the full name.

In other words, if built to the RFCs, DNS itself doesn't care if a DNS record has underscores or whether it's a valid "hostname" per se - just that it isn't too long. I have tested unbound and dnsmasq and they happily serve records with underscores but I'll update this PR to include the same change to dnsmasq's sync script.

**Comment by AdSchellevis** [comment]
@bugfixin thanks for testing and updating both. The restriction is there for good reasons, but that certainly doesn't mean we should not allow underscores if both applications do not mind. We'll take it under consideration.

**Comment by bugfixin** [comment]
Thank you - while per RFC 1123 extensibility section, DNS *should* allow any label containing up to 63 8-bit octets without care for validity, I agree that it makes sense to prevent certain characters that could cause problems.

I guess my host named 💩 will have to wait.

**Comment by AdSchellevis** [comment]
@bugfixin discussed internally, we'll merge this one. Thanks!
